### PR TITLE
feat: Allow for bodyDef and fixtureDefs to be prepared earlier

### PR DESCRIPTION
**Pull request by l1553k**
# Description
This change introduces a way to create `BodyDef` and `FixtureDef`s as early as in constructor - or even earlier. This allows to pass position and other body/fixture related parameters to constructor, and not have them stick around as `this.initialThis` or `this.initialThat`.

For example:
```
class OffroadWheel extends BodyComponent {
  OffroadWheel(
    Vector2 initialPosition,
    double radius,
  ) : super(
    bodyDef: BodyDef(position: initialPosition), type: BodyType.dynamic),
    fixtureDef: FixtureDef(CircleShape()..radius = radius),
  );
}
```

I added `fixtureDef` constructor argument for convenience.

## Checklist

- [x] I have followed the [Contributor Guide] when preparing my PR.
- [x] I have updated/added tests for ALL new/updated/fixed functionality.
- [x] I have updated/added relevant documentation in `docs` and added dartdoc comments with `///`.
- [x] I have updated/added relevant examples in `examples` or `docs`.

## Breaking Change?

- [x] No, this PR is not a breaking change.

## Related Issues

[Contributor Guide]: https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md
[Conventional Commit]: https://conventionalcommits.org
[CHANGELOG]: https://github.com/flame-engine/flame/blob/main/CHANGELOG.md

### CODE REVIEW

**Comment by l1553k** [comment]
@spydon we spoke about this on discord recently. Please provide some initial feedback; if you think this is going in the right direction then I will work on making it merge worthy.

File: packages/flame_forge2d/lib/body_component.dart [hunk 1]
```diff
@@ -21,14 +21,27 @@ abstract class BodyComponent<T extends Forge2DGame> extends Component
     super.children,
     super.priority,
     this.renderBody = true,
+    this.bodyDef,
+    FixtureDef? fixtureDef,
```
Comment (spydon) [inline packages/flame_forge2d/lib/body_component.dart:25]: I think you can skip this one and only have the list of fixture definitions

**Review by spydon — COMMENTED** [review]
Other than the analyze warnings and the comment I made I think it looks good!

**Comment by spydon** [comment]
Just these small things left and I think we're ready to merge:
* Added docs in the flame_forge2d section
* A line about the new arguments in the `BodyComponent` dartdocs 
* Some basic test

**Comment by l1553k** [comment]
> * A line about the new arguments in the `BodyComponent` dartdocs

I'm sorry for my noob question, but what do you mean?

**Comment by spydon** [comment]
> > * A line about the new arguments in the `BodyComponent` dartdocs
> 
> I'm sorry for my noob question, but what do you mean? 

The dartdocs are the ones that start with `///` above the class, so if you could just add two lines saying something like:
```
/// You can use the optional [bodyDef] and [fixtureDefs] arguments to create the [BodyComponent]'s body without having to create the definitions within the component.
```

File: doc/bridge_packages/flame_forge2d/forge2d.md [hunk 2]
```diff
@@ -51,8 +51,13 @@ A simple `Forge2DGame` implementation example can be seen in the
...
 The `BodyComponent` is a wrapper for the `Forge2D` body, which is the body that the physics engine
-is interacting with. To create a `BodyComponent` you need to override `createBody()` and create and
-return your created body.
+is interacting with. To create a `BodyComponent` you can either:
+
+- override `createBody()` and create and return your created body;
+- use the default `createBody()` implementation: pass a `BodyDef` instance (and optionally a list
+of `FixtureDef` instances) to BodyComponent' constructor' `bodyDef` (and `fixtureDefs`) arguments;
+- use the default `createBody()` implementation: assign a `BodyDef` instance to `this.bodyDef`, and
+optionally a list of `FixtureDef` instances to `this.fixtureDefs`.
```
Comment (spydon) [inline doc/bridge_packages/flame_forge2d/forge2d.md:60]:
```suggestion
- use the default `createBody()` implementation by passing a `BodyDef` instance (and optionally a
list of `FixtureDef` instances) to the BodyComponent's constructor;
- use the default `createBody()` implementation and assign a `BodyDef` instance to `this.bodyDef`,
and optionally a list of `FixtureDef` instances to `this.fixtureDefs`.
```

File: packages/flame_forge2d/lib/body_component.dart [hunk 3]
```diff
@@ -10,7 +10,11 @@ import 'package:flutter/foundation.dart';
 /// Since a pure BodyComponent doesn't have anything drawn on top of it,
 /// it is a good idea to turn on [debugMode] for it so that the bodies can be
 /// seen
-abstract class BodyComponent<T extends Forge2DGame> extends Component
+///
+/// You can use the optional [bodyDef] and [fixtureDefs] arguments to create
+/// the [BodyComponent]' body without having to create the definitions within
```
Comment (spydon) [inline packages/flame_forge2d/lib/body_component.dart:15]:
```suggestion
/// the [BodyComponent]'s body without having to create the definitions within
```

File: packages/flame_forge2d/lib/body_component.dart [hunk 4]
```diff
@@ -29,6 +35,15 @@ abstract class BodyComponent<T extends Forge2DGame> extends Component
   static const defaultColor = Color.fromARGB(255, 255, 255, 255);
   late Body body;
 
+  /// Default implementation of [createBody] will use this value, if provided.
```
Comment (spydon) [inline packages/flame_forge2d/lib/body_component.dart:38]:
```suggestion
  /// The default implementation of [createBody] will use this value to create the [Body],
  /// if it is provided.
```

File: packages/flame_forge2d/lib/body_component.dart [hunk 5]
```diff
@@ -29,6 +35,15 @@ abstract class BodyComponent<T extends Forge2DGame> extends Component
...
+  ///
+  /// If you do not provide a BodyDef here, you must override [createBody].
```
Comment (spydon) [inline packages/flame_forge2d/lib/body_component.dart:40]:
```suggestion
  /// If you do not provide a [BodyDef] here, you must override [createBody].
```

File: packages/flame_forge2d/lib/body_component.dart [hunk 6]
```diff
@@ -29,6 +35,15 @@ abstract class BodyComponent<T extends Forge2DGame> extends Component
...
+  BodyDef? bodyDef;
+
+  /// Default implementation of [createBody] adds these fixtures to the body
+  /// that it creates from [bodyDef].
```
Comment (spydon) [inline packages/flame_forge2d/lib/body_component.dart:44]:
```suggestion
  /// The default implementation of [createBody] will add these fixtures to the [Body] that it
  /// creates from [bodyDef].
```

**Review by spydon — APPROVED** [review]

**Review by spydon — COMMENTED** [review]
Fix line length

**Pull request merged**